langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -868,6 +868,45 @@ function testFromJsonWithTypeTable() {
 
 }
 
+type IntVal record { int? x;};
+
+type Cases record {
+    Case[] Data;
+    string LastData;
+    string Source;
+    string DevBy;
+};
+
+type Case record {
+  string ConfirmDate;
+  string No;
+  int? Age;
+  string Nation;
+  string? Detail;
+};
+
+function testFromJsonWithTypeWithNullValues() {
+    json j1 = {x: null};
+    IntVal val = checkpanic j1.fromJsonWithType(IntVal);
+    assert(val.toString(), "{\"x\":null}");
```

Comment:
It's generally better to assert the actual value rather than checking `toString` since we may lose information about types when asserting string representations.